tests.py

```diff
@@ -372,6 +373,23 @@ def test_uri_template_matcher_duplicate_variable_error():
         UriTemplateMatcher(u'/foo/{var}/bar/{var}')
 
 
+@mark.parametrize('lval, rval, expected', [
+    (None, [('n1', 'v1'), ('n2', 'v2')], [('n1', 'v1'), ('n2', 'v2')]),
+    ([('n1', 'v1'), ('n2', 'v2')], None, [('n1', 'v1'), ('n2', 'v2')]),
+    ([('n1', 'v1'), ('n2', 'v2')], [('n1', 'v3'), ('n2', 'v4')],
+     [('n1', 'v1'), ('n2', 'v2'), ('n1', 'v3'), ('n2', 'v4')]),
+    (None, None, None),
+])
+def test_uri_template_match_result_update(lval, rval, expected):
+    lval = UriTemplateMatchResult(lval)
+    rval = UriTemplateMatchResult(rval)
+    lval.update(rval)
+    if expected is None:
+        assert not lval
```

Comment:
`not lval` is enough to pass a test. But if `lval` has to be `None` when `expected` is `None`, it is better to test using `is` statement like `lval is None`.